[FIX] account: invoice report duplicated values

niyasraphy opened the pull request:
**Description of the issue/feature this PR addresses:**
Duplicated values in the invoice report. 
Terms and Conditions, Fiscal Position, Incoterms and QR Code

**Current behavior before PR:**
Duplicated values in invoice report.

**Desired behavior after PR is merged:**
Remove duplicated values




---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/103878).

jwaes commented on addons/account/views/report_invoice.xml:
```diff
@@ -250,21 +250,6 @@
                         <p t-if="o.invoice_payment_term_id" name="payment_term">
                             <span t-field="o.invoice_payment_term_id.note"/>
                         </p>
```
These lines should also be removed i guess

niyasraphy replied:
Thanks for pointing it, this too seems duplicated.
updated the pr

niyasraphy commented:
@Levizar @FlorianGilbert 
If there is some time, can you review this PR ? currently multiple values are duplicated in the invoice print out

Levizar commented:
Hi @casp-odoo Could you have a look to this? It's removing some code you recently introduced and I'm worrying that it could "break" what you've made.

Levizar commented:
ping @ZedarkMoon

casp-odoo commented:
LGTM, it indeed seems uselessly duplicated

niyasraphy commented:
> Hi @casp-odoo Could you have a look to this? It's removing some code you recently introduced and I'm worrying that it could "break" what you've made.
> 
> Edit: it seems that SMDC already had a look and proposed this fix instead: #103890



> Hi @casp-odoo Could you have a look to this? It's removing some code you recently introduced and I'm worrying that it could "break" what you've made.
> 
> Edit: it seems that SMDC already had a look and proposed this fix instead: #103890

actually there more duplicated fields than in this fix

niyasraphy commented:
@Levizar actually there are more duplicated values than in this fix: https://github.com/odoo/odoo/pull/103890
Included all in this PR: https://github.com/odoo/odoo/pull/103878

ZedarkMoon commented:
@Levizar @casp-odoo Indeed multiple duplicate values are present when QR code option is ticked and therefor can be remove. I'm in favor of this fix #103878 to remove them and keep those informations outside of the QR code part because, in my opinion, I don't see why the payment terms, terms and condition, fiscal position note... should only appear when the option is active.
Already close #103890 and I've added a link to this PR.

FlorianGilbert reviewed (approved):
@robodoo r+

robodoo commented:
@niyasraphy @FlorianGilbert staging failed: ci/l10n on f0d7b1621c9d1fbb20fe35be694a2b2fbdc4361b (view more at https://runbot.odoo.com/runbot/build/20494717)

niyasraphy commented:
@FlorianGilbert is there anything to do from my side ?

casp-odoo commented:
https://github.com/odoo/odoo/pull/103878/files#diff-c85e75bc27fc841662ca0598c09a22670e0a0b4e5120815934c0a50999ff02bfL256

The `note` field is used in the l10n_it xpath, and it is removed in your PR, hence one of the errors. 

I guess the other ones just need a rebuild

niyasraphy commented:
@casp-odoo thanks for pointing the issue.
I need a suggestion from your side, in account module initially we were having a p and div with name notes and i have removed the p with name note as the same content is displayed by the div.

In l10_it module is looking for xpath of the remove p, should i change this xpath to check for div instead of p or should i change the original div to p ?
Looking for your thoughts

niyasraphy commented:
@casp-odoo i updated PR with changing the l10_it module to look for div with name note instead of p

lordkrandel commented:
The italian code just needs to override the note in some condition.

In retrospective, we don't care if it's `p` or `div` as long as the target tag has the `name` attribute and contains the note.

So personally I would change the selector to something like:

`*[name="note"]`

without specifying the tag name.

niyasraphy commented:
@lordkrandel Thanks,
I already made the patch with changing p to div, should i update it as you suggested or keep the current patch ?

FlorianGilbert commented:
@robodoo delegate=@casp-odoo

casp-odoo commented:
@robodoo r+

robodoo commented:
@niyasraphy @casp-odoo staging failed: ci/l10n on b8149f65ff43df5e941f64e24d47f513c25747c8 (view more at https://runbot.odoo.com/runbot/build/20604269)

niyasraphy commented:
updated the pr again as the xpath failed in the module l10n_mx_edi, similar to what happens with l10_it module earlier.
As l10n_mx_edi is part of EE, i have updated with reverting the changes made in l10_it module and updated the original layout with p instead of div.

@casp-odoo @lordkrandel

casp-odoo commented:
@robodoo r+
